From 3378f7c984df9a2c626a3f2ff14f7c53b607f3ff Mon Sep 17 00:00:00 2001
From: Justin Hammond <justin@dynam.ac>
Date: Wed, 22 Feb 2023 16:25:32 +0800
Subject: [PATCH] add default script address

---
 include/configs/bl808.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/bl808.h b/include/configs/bl808.h
index 7fe9d1d165..976640894d 100644
--- a/include/configs/bl808.h
+++ b/include/configs/bl808.h
@@ -45,6 +45,7 @@
 	BOOT_TARGET_DEVICES_DHCP(func)
 
 #define CFG_EXTRA_ENV_SETTINGS \
+        "scriptaddr=0x50000000\0" \
 	BOOTENV
 #endif
 
-- 
2.25.1

